ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/transform/PubsubMessageToObjectNode.java

```diff
@@ -344,10 +344,13 @@ private void processField(String jsonFieldName, Field field, JsonNode value, Obj
           updateParent(parent, name, processTupleField(jsonFieldName, field.getSubFields(),
               (ArrayNode) value, additionalProperties));
         } else {
-          final ObjectNode props = additionalProperties == null ? null : Json.createObjectNode();
-          transformForBqSchema((ObjectNode) value, field.getSubFields(), props);
-          if (!Json.isNullOrEmpty(props)) {
-            additionalProperties.set(jsonFieldName, props);
+          // Only transform value if it is not null
+          if (!value.isNull()) {
+            final ObjectNode props = additionalProperties == null ? null : Json.createObjectNode();
```

Comment:
```suggestion
            final ObjectNode props = (additionalProperties == null) ? null : Json.createObjectNode();
```

I find this easier to parse with parentheses.